test.array.js

```diff
@@ -0,0 +1,25 @@
+var assert = require('assert');
+
+var fakeArray = [];
+
+describe('My fake array object', function() {
+  describe('The pop method', function() {
+
+    before(function() {
+      fakeArray[0] = 1;
+      fakeArray[1] = 2;
+      fakeArray[2] = 3;
+      fakeArray.length = 3;
+    });
+
+    it('should return the final element', function() {
+      assert.equal(fakeArray.pop(), 3);
```

Comment:
...and with a few caveats in the runner below, we can rewrite this assertion using chai's `expect()` syntax as:

    expect(fakeArray.pop()).to.eq(3);